### PR TITLE
Upload of quick_template and re-structured decbinhex4.

### PR DESCRIPTION
**Pull request by insomniacUNDERSCORElemon**
Also instructions in readme, a fix to decbinhex4 live-binary-to-decimal.

### CODE REVIEW

File: decbinhex4/LICENSE
Comment (panno8M) [inline decbinhex4/LICENSE:1]: Is it okay?

File: decbinhex4/core/main.gdextension [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+entry_symbol = "init_library"
+compatibility_minimum = 4.1
+
+[libraries]
+
+linux.debug.x86_64 = "res://core/lib/libcore.so"
+linux.release.x86_64 = "res://core/lib/libcore.so"
+windows.debug.x86_64 = "res://core/lib/libcore.dll"
+windows.release.x86_64 = "res://core/lib/libcore.dll"
```
Comment (panno8M) [inline decbinhex4/core/main.gdextension]: I recently learned that apparently this section works with a more abstract assignment.

```
linux = "res://core/lib/libcore.so"
windows = "res://core/lib/libcore.dll"
```

You may leave the designation of debug or release for future use. Perhaps a specification about CPU architecture is not necessary.

File: decbinhex4/config.nims [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## This is applied to all workspaces below it, thus applies to both extensions.
+import os
+
+--app: lib
+--noMain: on
+--define: nimPreviewDotLikeOps
+
+let workspaceName = projectDir().splitFile.name
+
+switch("define", "projectName:" & workspaceName)
```
Comment (panno8M) [inline decbinhex4/config.nims]: projectName: has been changed to ExtensionName: in the latest version.

```nim
switch("define", "ExtensionName:" & workspaceName)
```

**Review by panno8M — CHANGES_REQUESTED** [review]

File: decbinhex4/LICENSE
Comment (insomniacUNDERSCORElemon) [reply decbinhex4/LICENSE:1]: Whoops, I forgot to make that file on my local machine. EDIT: Also on the topic of licenses, note the issue I opened in this repo.

File: decbinhex4/config.nims (hunk 2 above)
Comment (insomniacUNDERSCORElemon) [reply decbinhex4/config.nims]: I still need to `let` workspaceName though, I assume?

**Comment by insomniacUNDERSCORElemon** [comment]
Any other changes, thoughts?

**Comment by insomniacUNDERSCORElemon** [comment]
Any other changes, thoughts?

File: addon_template/addons/extension/config.nims
Comment (panno8M) [inline addon_template/addons/extension/config.nims:1]: This file is still out of date.

File: quick_template/core/config.nims [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# begin Nimble config (version 2)
+when withDir(thisDir(), system.fileExists("nimble.paths")):
+  include "nimble.paths"
+# end Nimble config
+
+# begin gdext spacification
+when withDir(thisDir(), system.fileExists("bootstrapconf.nims")):
+  include "bootstrapconf.nims"
+# end gdext spacification
```
Comment (panno8M) [inline quick_template/core/config.nims:21]: This section is not necessary for this configuration.

File: addon_template/addons/extension/extension.gdextension
Comment (panno8M) [inline addon_template/addons/extension/extension.gdextension:1]: same

File: decbinhex4/config.nims [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# begin Nimble config (version 2)
+when withDir(thisDir(), system.fileExists("nimble.paths")):
+  include "nimble.paths"
+# end Nimble config
+
+# begin gdext spacification
+when withDir(thisDir(), system.fileExists("bootstrapconf.nims")):
+  include "bootstrapconf.nims"
+# end gdext spacification
+
```
Comment (panno8M) [inline decbinhex4/config.nims:22]: same

**Review by panno8M — CHANGES_REQUESTED** [review]
.gitignore and .gitattributes are not in decbinhex4 and quick_template.

File: addon_template/addons/extension/config.nims
Comment (insomniacUNDERSCORElemon) [reply addon_template/addons/extension/config.nims:1]: Can you update that project to your intended publishing structure? I was just renaming the root folder so there wasn't a name conflict (/for clarity).

**Comment by insomniacUNDERSCORElemon** [comment]
> .gitignore and .gitattributes are not in decbinhex4 and quick_template.

That was because dotfiles don't upload with the project for whatever reason. But I can manually add them back.

File: quick_template/core/config.nims (hunk 3 above)
Comment (insomniacUNDERSCORElemon) [reply quick_template/core/config.nims:21]: I had that thought, but is this an issue? It's a template, someone might end up creating that file (if it is still in spec) especially if they use multiple extensions so I don't see the point in trimming it out.

**Comment by insomniacUNDERSCORElemon** [comment]
You didn't give me enough time to upload the .gitignore etc, I would've marked it as resolved if I had. EDIT: I'll do a new PR w/it.

**Comment by panno8M** [comment]
It's a bad habit of mine. There were a lot of other changes, so I thought I'd merge them in once and fix them here. I should have said so.

**Pull request merged**